osu.Game/Screens/Play/HUD/GameplaySpectatorList.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Configuration;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Localisation;
+using osu.Game.Users;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Screens.Play.HUD
+{
+    public partial class GameplaySpectatorList : CompositeDrawable
+    {
+        private readonly Bindable<bool> configVisibility = new Bindable<bool>();
+
+        // This is going to dynamically change depending on the amount of spectators
+        protected readonly FillFlowContainer<GameplaySpectatorUser> Flow;
+        protected readonly FillFlowContainer OtherSpectators;
+        protected OsuSpriteText SpectatorText;
+        private const int fade_time = 100;
+
+        public int Spectators => Flow.Count;
+        public readonly BindableInt MaxNames = new BindableInt(10);
```

Comment:
Why is this bindable?

Reply:
To facilitate (pending discussion about design, both UI + UX) to change the amount of users shown, possibly another setting somewhere. This was sort-of a left-over from writing the tests, to make sure the fading works as intended.